wom/src/main/scala/wom/graph/CallNode.scala

```diff
@@ -106,14 +106,14 @@ object CallNode {
     }
   }
 
-  final case class InputDefinitionFold(mappings: InputDefinitionMappings = Map.empty,
+  final case class InputDefinitionFold(mappings: InputDefinitionMappings = List.empty,
                                        callInputPorts: Set[InputPort] = Set.empty,
                                        newGraphInputNodes: Set[ExternalGraphInputNode] = Set.empty,
                                        newExpressionNodes: Set[ExpressionNode] = Set.empty,
                                        usedOuterGraphInputNodes: Set[OuterGraphInputNode] = Set.empty)
 
   type InputDefinitionPointer = OutputPort :+: WomExpression :+: WomValue :+: CNil
-  type InputDefinitionMappings = Map[InputDefinition, InputDefinitionPointer]
+  type InputDefinitionMappings = List[(InputDefinition, InputDefinitionPointer)]
```

Comment:
Please leave a comment as to why one shouldn't come back and refactor this into a `Map`.

ToL: After our chat about `Set`s (including key sets) yesterday, I'm more than happy to have this as a list, but I'm curious as to if this was a bug fix, or a preference fix. If this is a bug fix, do the centaur tests fix this if it's a regression? Someday I'd love to also see more unit tests in this area too.

Reply:
Yes, there's a centaur test called `optional_parameter` that was failing before this change was made.

Reply:
I worry about having centaur tests that exist to test other things and only incidentally test the important functionality here. i.e. those tests might get refactored and we wouldn't know if this was still working. tl;dr can we have an explicit test

Reply:
I dunno whether I'd agree about it being incidental... the only reason this is a `List` is so that we can evaluate inputs in the correct order. 

If someone somehow refactors this and that test still works... I'm quite happy to let that happen

Reply:
`select_first_with_optionals` is an explicit test of using optional inputs in subsequent expressions, so I don't think it's likely to be refactored. 

To make it even less likely I've added a comment to preserve that feature of the test.